src/snapred/backend/recipe/algorithm/MantidSnapper.py

```diff
@@ -148,7 +144,9 @@ def executeAlgorithm(self, name, outputs, **kwargs):
                     val = val.get()
                 if val is None:
                     continue
-
+                # turn pydantic objects into pointers
+                if isinstance(algorithm.getProperty(prop), PointerProperty):
+                    val = create_pointer(val)
```

Comment:
do we want to type check `val` as well and ensure its actually a pydantic object?  or should the comment be updated to generically refer to pyobjects?

Reply:
Val should be any python object, not limited to pydantic